src/attributes.jl

```diff
@@ -250,7 +250,8 @@ end
 function Base.setindex!(attrdict::AttributeDict, val, name::AbstractString)
     if haskey(attrdict, name)
         # in case of an error, we write first to a temporary, then rename
-        _name = tempname()
+        _name = "-><7;*<_NY" * bytes2hex(rand(Random.RandomDevice(), UInt8, 16))
```

Comment:
What's the rationale of the initial string? In other words, why not rely on the bytes2hex(' random number string).

Reply:
I initially had that to make collisions even less likely, but trying random names multiple times is probably even better.